### PR TITLE
[CAMARA] - Release on the prod

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Release on the prod

**Youtrack:** https://youtrack.raccoongang.com/agiles/104-17/105-19?issue=CAMARA-82

**Settings**:
AUTH_TOKENS (lms.auth.json):

CC_PROCESSOR = {
...
'MPesa':{
"PURCHASE_ENDPOINT": '/mpesa/payment/',
}
....
}

ENV_TOKENS (lms.env.json):
TOKEN_THRESHOLD = 600

Add in admin page in site configuration. As Example for Kenya microsite

CC_PROCESSOR_NAME":"MPesa"
MPESA_B2C_ACCESS_KEY :"cUbohz0F5XC49DcDNrd9gdSjlBC0JFGi"
MPESA_B2C_CONSUMER_SECRET : "b1eqpdYroaWiJAlG"
MPESA_C2B_ACCESS_KEY : "cUbohz0F5XC49DcDNrd9gdSjlBC0JFGi"
MPESA_C2B_CONSUMER_SECRET : "b1eqpdYroaWiJAlG"
C2B_REGISTER_URL = 'https://sandbox.safaricom.co.ke/mpesa/c2b/v1/registerurl'
B2C_INITIATOR_NAME : 'testapi409'
B2C_COMMAND_ID : 'SalaryPayment'
B2C_SHORTCODE : '545980'
B2C_QUEUE_TIMEOUT_URL : 'https://kenya.learn.camara.org/mpesa/b2c/timeout'
B2C_RESULT_URL : 'https://kenya.learn.camara.org/mpesa/b2c/result'
C2B_VALIDATE_URL : 'https://kenya.learn.camara.org/mpesa/c2b/validate'
C2B_CONFIRMATION_URL : 'https://kenya.learn.camara.org/mpesa/c2b/confirmation'
C2B_SHORT_CODE : '545980'
C2B_RESPONSE_TYPE : 'ok'
C2B_ONLINE_CHECKOUT_URL = 'https://sandbox.safaricom.co.ke/mpesa/stkpush/v1/processrequest'
C2B_ONLINE_CHECKOUT_CALLBACK_URL : 'https://kenya.learn.camara.org/mpesa/c2b/online_checkout/{}/callback'
C2B_TRANSACTION_TYPE: 'CustomerPayBillOnline'
C2B_ONLINE_PASSKEY : 'bfb279f9aa9bdbcf158e97dd71a467cd2e0c893059b10f78e6b72ada1ed2c919'
C2B_ONLINE_SHORT_CODE : '174379'
GENERATE_TOKEN_URL = 'https://sandbox.safaricom.co.ke/oauth/v1/generate?grant_type=client_credentials'
B2C_URL = 'https://sandbox.safaricom.co.ke/mpesa/b2c/v1/paymentrequest'

### CODE REVIEW

File: lms/djangoapps/shoppingcart/processors/MPesa.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    Returns:
+        dict
+
+    """
+    total_cost = cart.total_cost
+    amount = "{0:0.2f}".format(total_cost)
+    params = OrderedDict()
+    #TODO: add all params for Mpesa API
+    return params
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/processors/MPesa.py]: Trailing empty line is missed.

File: lms/djangoapps/shoppingcart/processors/MPesa.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        callback_url (unicode): The URL that CyberSource should POST to when the user
+            completes a purchase.  If not provided, then CyberSource will use
+            the URL provided by the administrator of the account
+            (CyberSource config, not LMS config).
+
+        extra_data (list): Additional data to include as merchant-defined data fields.
+
+    Returns:
+        dict
+
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/processors/MPesa.py]: An odd empty line

File: lms/djangoapps/shoppingcart/processors/MPesa.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        log.error('Order "{}" and transaction "{}" is failed'.format(order, mpesa_result['txn_id']))
+        return {'success': False, 'order': order, 'error_html': 'Transaction "{}" is filed'.format(mpesa_result['txn_id'])}
+
+def get_purchase_endpoint():
+    """
+    Return the URL of the payment end-point for CyberSource.
+
+    Returns:
+        unicode
+
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/processors/MPesa.py]: An odd empty line could be omitted.

File: lms/envs/aws.py [hunk 4]
```diff
@@ -1043,3 +1043,36 @@
...
+C2B_TRANSACTION_TYPE = AUTH_TOKENS.get('C2B_TRANSACTION_TYPE', '')
+C2B_ONLINE_PASSKEY = AUTH_TOKENS.get('C2B_ONLINE_PASSKEY', '')
+C2B_ONLINE_SHORT_CODE = AUTH_TOKENS.get('C2B_ONLINE_SHORT_CODE', '')
+
+# Urls
+GENERATE_TOKEN_URL = AUTH_TOKENS.get('GENERATE_TOKEN_URL', '')
+B2C_URL = AUTH_TOKENS.get('B2C_URL', '')
+
+# number of seconds from the expiry we consider the token expired
+TOKEN_THRESHOLD = ENV_TOKENS.get('TOKEN_THRESHOLD', 600)
```
Comment (idegtiarov) [inline lms/envs/aws.py]: Trailing empty line is missed. Please add it.

File: lms/envs/common.py [hunk 5]
```diff
@@ -3227,4 +3231,4 @@
 
 ############## Settings for CourseGraph ############################
 COURSEGRAPH_JOB_QUEUE = LOW_PRIORITY_QUEUE
-ENABLE_ENTERPRISE_INTEGRATION = False
+ENABLE_ENTERPRISE_INTEGRATION = False
```
Comment (idegtiarov) [inline lms/envs/common.py]: ditto

File: lms/envs/devstack.py [hunk 6]
```diff
@@ -117,15 +117,36 @@ def should_show_debug_toolbar(_):
 ########################### VERIFIED CERTIFICATES #################################
 
 FEATURES['AUTOMATIC_VERIFY_STUDENT_IDENTITY_FOR_TESTING'] = True
-FEATURES['ENABLE_PAYMENT_FAKE'] = True
-
-CC_PROCESSOR_NAME = 'CyberSource2'
+FEATURES['ENABLE_PAYMENT_FAKE'] = False
+
+# CC_PROCESSOR_NAME = 'CyberSource2'
```
Comment (idegtiarov) [inline lms/envs/devstack.py]: Why do we need these commented lines?

File: lms/djangoapps/course_api/serializers.py [hunk 7]
```diff
@@ -96,6 +101,26 @@ def get_blocks_url(self, course_overview):
...
+        site_currency = configuration_helpers.get_value('PAID_COURSE_REGISTRATION_CURRENCY')
+
+        if site_currency:
+            registration_price = CourseMode.min_course_price_for_currency(course_overview.id, site_currency[0])
+            currency = site_currency[0]
+        else:
+            registration_price, currency = CourseMode.min_course_price(course_overview.id)
+
+        if registration_price:
+            currency_symbol = dict(settings.CURRENCIES).get(currency, settings.PAID_COURSE_REGISTRATION_CURRENCY[0])
```
Comment (idegtiarov) [inline lms/djangoapps/course_api/serializers.py]: I can not find where param `CURRENCIES ` is set up in the settings files. Do we need to specify it?

**Comment by AndreyLykhoman** [comment]
I fixed mistakes in this PR: https://github.com/raccoongang/edx-platform/pull/536

File: lms/djangoapps/shoppingcart/processors/MPesa.py (hunk 1 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/shoppingcart/processors/MPesa.py]: Done. (https://github.com/raccoongang/edx-platform/pull/536)

File: lms/djangoapps/shoppingcart/processors/MPesa.py (hunk 2 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/shoppingcart/processors/MPesa.py]: Done. (https://github.com/raccoongang/edx-platform/pull/536)

File: lms/djangoapps/shoppingcart/processors/MPesa.py (hunk 3 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/shoppingcart/processors/MPesa.py]: Done (https://github.com/raccoongang/edx-platform/pull/536)

File: lms/envs/aws.py (hunk 4 above)
Comment (AndreyLykhoman) [reply lms/envs/aws.py]: Added. (https://github.com/raccoongang/edx-platform/pull/536)

File: lms/envs/common.py (hunk 5 above)
Comment (AndreyLykhoman) [reply lms/envs/common.py]: Added(https://github.com/raccoongang/edx-platform/pull/536)

File: lms/envs/devstack.py (hunk 6 above)
Comment (AndreyLykhoman) [reply lms/envs/devstack.py]: I forgot to delete these lines. I deleted it. (I forgot to delete these lines)

File: lms/djangoapps/course_api/serializers.py (hunk 7 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/course_api/serializers.py]: These settings already set in production server. The task that related to it: https://youtrack.raccoongang.com/issue/CAMARA-51 
Also, I updated my comment in this task https://youtrack.raccoongang.com/agiles/104-17/105-19?issue=CAMARA-82 I wrote about check these setting.

**Pull request merged**